magento/pwa-studio
# Use graphql to add item to cart

## Description

* Updates `addItemToCart` to pass mutations for adding different product types to the cart.

* Updates `updateItemInCart` to pass the necessary mutations for add item since addItem action was used as a fallback case.

* Updates retry logic for graphQL response in the addItem error catch.

## Related Issue





Closes [PWA-139](https://jira.corp.magento.com/browse/PWA-139).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Add a simple product to cart (no options ie jewelry). The cart should display properly with the item.
2. Add a configurable product to cart (a dress or something). The cart should display properly with the item.
3. Log in and add an item to the cart. It should display properly.
3. Log out and add an item to the cart. It should display properly.
4. Edit an item in the cart. It should update properly.
5. Remove an item from the cart. It should be removed properly.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
